pallets/staking/src/pallet/impls.rs

```diff
@@ -889,13 +889,14 @@ impl<T: Config> Pallet<T> {
             };
 
             if let Some(Nominations { targets, .. }) = <Nominators<T>>::get(&voter) {
+                if !targets.is_empty() {
+                    nominators_seen.saturating_inc();
+                }
```

Comment:
Since our `is_nominator_compliant` check is going to be the biggest cost (multiple reads) per nominator, we should increment this counter even when `targets` is empty.